api/services/user_contact_service.py

```diff
@@ -86,10 +96,27 @@ def process_custom_fields(self, jsonData, user_custom_field_data=False):
             ):
                 userdata = self.get_user_custom_fields_object(field_name, field_value)
                 user_custom_contact_data.extend(userdata)
-            elif custom_fields_conditions:
+            elif custom_fields_conditions and not self.check_if_exist(
+                active_custom_fields_key_values, field_name, field_value
+            ):
                 models.UserCustomFields.query.set_custom_field_as_active(
                     field_name, field_value, self.user_phone
                 )
+            elif custom_fields_conditions and self.check_if_exist(
+                active_custom_fields_key_values, field_name, field_value
+            ):
+                active_custom_fields_key_values.pop(
+                    str(field_name) + "_" + str(field_value)
+                )
+
+        for (
+            custom_field_name,
+            custom_field_value,
+        ) in active_custom_fields_key_values.items():
+            custom_field_key = custom_field_name[: -(len(custom_field_value) + 1)]
+            models.UserCustomFields.query.set_custom_field_as_inactive(
+                custom_field_key, custom_field_value, self.user_phone
+            )
```

Comment:
Marking the custom fields as inactive that are currently active in the database.